### ISSUE TITLE
librte_malloc has been moved to eal/common in DPDK. Seems no librte_malloc anymore.

### ISSUE THREAD

**Issue by changhuilin** [spdk/spdk#3]
I am using the current version of DPDK from github.
Since there is no librte_malloc, in mk/spdk.common.mk,  -lrte_malloc needs to be removed from DPDK_LIB for new versions of dpdk.

**Comment by dverkamp-intel** [comment]
It appears that the functionality from the rte_malloc library has been merged into the core rte_eal library as of DPDK 2.1, and DPDK 2.2 removes the empty compatibility librte_malloc: http://dpdk.org/browse/dpdk/commit/?id=53dbab96d5ecccd7662f4de3c17030ea3e11cda2

We will have to either declare incompatibility with DPDK 2.0 and older or add some compile-time detection to link against the right libraries.

**Comment by chrisvest** [comment]
Meanwhile, this change appears to work, in the sense that the unit tests appears to pass:

```diff
diff --git a/mk/spdk.common.mk b/mk/spdk.common.mk
index 03b0d62..26d8d2a 100644
--- a/mk/spdk.common.mk
+++ b/mk/spdk.common.mk
@@ -118,7 +118,7 @@ DPDK_INC_DIR ?= $(DPDK_DIR)/include
 DPDK_LIB_DIR ?= $(DPDK_DIR)/lib
 
 DPDK_INC = -I$(DPDK_INC_DIR)
-DPDK_LIB = -L$(DPDK_LIB_DIR) -lrte_eal -lrte_malloc -lrte_mempool -lrte_ring -Wl,-rpath=$(DPDK_LIB_DIR)
+DPDK_LIB = -L$(DPDK_LIB_DIR) -lrte_eal -lrte_mempool -lrte_ring -Wl,-rpath=$(DPDK_LIB_DIR)
 # DPDK requires dl library for dlopen/dlclose on Linux.
 ifeq ($(OS),Linux)
 DPDK_LIB += -ldl
```

(just removing the `-lrte_malloc` from the `DPDK_LIB` line)

Please let me know if it's somehow a wrong way to work around this problem.

**Comment by changhuilin** [comment]
Yes, for the newer versions of DPDK, removing -lrte_malloc fixes the problem; but this breaks the use of older versions of DPDK. This is a compatibility issue, so I think the best way is to somehow have a compile-time detection.

**Comment by dverkamp-intel** [comment]
Fixed by @jimharris in 38ea764f6da631d31e3d1072a1a526c2eab6a214 - librte_malloc is detected at build time, so SPDK will now link with DPDK <= 2.1 (with librte_malloc) and DPDK >= 2.2 (without librte_malloc).
